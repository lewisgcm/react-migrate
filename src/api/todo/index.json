[
    {
        "userId": 1,
        "id": 1,
        "title": "clean the dishes",
        "completed": false
    },
    {
        "userId": 2,
        "id": 2,
        "title": "wash the floors",
        "completed": false
    },
    {
        "userId": 3,
        "id": 3,
        "title": "take out the rubbish",
        "completed": false
    }
]
