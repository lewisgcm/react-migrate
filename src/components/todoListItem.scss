.TodoListItem {
    width: 100%;
    line-height: 1.8em;

    span {
        border: 1px solid red;
        background-color: #FF8888;
        border-radius: 15px;
        height: 24px;
        width: 24px;
        text-align: center;
        font-size: .6em;
        float: right;
        cursor: pointer;
    }

    span:hover {
        background-color: #ffeeeeee;
    }
}
