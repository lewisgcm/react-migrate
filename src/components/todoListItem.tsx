import * as React from "react";

import { Todo } from "../models/todo";

const style = require( "./todoListItem.scss" );

interface TodoListItemProps {
    todo: Todo;
    onDeleteTodo: (todo: Todo) => void;
}

interface TodoListItemState {
    todo: Todo;
}

export class TodoListItem extends React.Component<TodoListItemProps, TodoListItemState> {

    constructor( props: TodoListItemProps ) {
        super( props );
        this.state = {
            todo: props.todo,
        };
    }

    public render() {
        return <li className={style.TodoListItem}>
            {this.state.todo.title}
            <span onClick={( e ) => this.handleOnDelete()}>X</span>
        </li>;
    }

    private handleOnDelete() {
        this.props.onDeleteTodo( this.state.todo );
    }
}
