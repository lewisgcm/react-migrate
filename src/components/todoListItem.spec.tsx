import * as React from "react";
import { create } from "react-test-renderer";

import { Todo } from "../models/todo";
import { TodoListItem } from "./todoListItem";

test( "Render todo list item", () => {
    const todo = {
        id: 0,
        userId: 1,
        title: "My todo",
        completed: true,
    } as Todo;

    const todoListItem = create( <TodoListItem todo={todo} onDeleteTodo={() => { /**/ }} /> );
    expect( todoListItem ).toMatchSnapshot();
});

test( "Delete todo list item", () => {
    const todo = {
        id: 0,
        userId: 1,
        title: "My todo",
        completed: true,
    } as Todo;

    const mockHandleDelete = jest.fn();
    const todoListItem = create( <TodoListItem todo={todo} onDeleteTodo={mockHandleDelete} /> );
    todoListItem.root.find( (e) => e.type === "span" ).props.onClick({});

    expect( todoListItem ).toMatchSnapshot();
    expect( mockHandleDelete ).toHaveBeenCalled();
});
