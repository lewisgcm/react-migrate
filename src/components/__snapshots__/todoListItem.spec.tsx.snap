// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete todo list item 1`] = `
<li
  className="TodoListItem"
>
  My todo
  <span
    onClick={[Function]}
  >
    X
  </span>
</li>
`;

exports[`Render todo list item 1`] = `
<li
  className="TodoListItem"
>
  My todo
  <span
    onClick={[Function]}
  >
    X
  </span>
</li>
`;
